bbl_screen-patch/patches/printerui/qml/X1Plus.js

```diff
@@ -119,10 +132,17 @@ X1Plus.formatTime = formatTime;
  * PrintManager passed down, and the real QML environment is truly alive. 
  * These things happen from 'awaken'.
  */
-function awaken(_DeviceManager, _PrintManager) {
+
+
+function awaken(_DeviceManager, _PrintManager, _PrintTask) {
 	console.log("X1Plus.js awakening");
 	X1Plus.DeviceManager = _DeviceManager;
 	X1Plus.PrintManager = _PrintManager;
+	X1Plus.PrintTask = _PrintTask;
 	BedMeshCalibration.awaken();
 	ShaperCalibration.awaken();
+	GpioKeys.awaken();
+	// X1Plus.isIdle = _PrintManager.currentTask.stage < _PrintTask.WORKING;
+	X1Plus.hasSleep = _DeviceManager.power.hasSleep;
+	X1Plus.aboutToSleep= _DeviceManager.power.aboutToSleep; 
```

Comment:
Ahhh, ok, so, yeah, this will not work.  This captures the immediate value, but does not capture a reference to the binding.  Basically this will capture whether `hasSleep` / `aboutToSleep` right when we `awaken`, but not at any point in the future.  I think the way is to make functions `X1Plus.hasSleep`, etc.